pkg/monitor/monitor/monitor_manager.go

```diff
@@ -427,3 +471,186 @@ func (m *MonitorManager) patchTidbClusterStatus(tcRef *v1alpha1.TidbClusterRef,
 	_, err = m.deps.Clientset.PingcapV1alpha1().TidbClusters(tc.Namespace).Patch(tc.Name, types.MergePatchType, mergePatch)
 	return err
 }
+
+func (m *MonitorManager) smoothMigrationToStatefulSet(monitor *v1alpha1.TidbMonitor) (bool, error) {
+	// determine whether there is an old deployment
+	oldDeploymentName := GetMonitorObjectName(monitor)
+	oldDeployment, err := m.deps.DeploymentLister.Deployments(monitor.Namespace).Get(oldDeploymentName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			if monitor.Spec.Persistent {
+				firstStsPvcName := GetMonitorFirstPVCName(monitor.Name)
+				deploymentPvcName := GetMonitorObjectName(monitor)
+				deploymentPvc, err := m.deps.PVCLister.PersistentVolumeClaims(monitor.Namespace).Get(deploymentPvcName)
+				if err != nil {
+					if errors.IsNotFound(err) {
+						// If the PVC of the deployment does not exist and no old pvc status, we don't need to migrate.
+						if monitor.Status.DeploymentStorageStatus != nil && len(monitor.Status.DeploymentStorageStatus.PvName) > 0 {
+							deploymentPv, err := m.deps.PVLister.Get(monitor.Status.DeploymentStorageStatus.PvName)
+							if err != nil {
+								klog.Errorf("Smooth migration for tm[%s/%s], fail to get PV %s, err: %v", monitor.Namespace, monitor.Name, monitor.Status.DeploymentStorageStatus.PvName, err)
+								return false, err
+							}
+							deploymentPv.Spec.ClaimRef.Name = firstStsPvcName
+							err = m.deps.PVControl.PatchPVClaimRef(monitor, deploymentPv, firstStsPvcName)
+							if err != nil {
+								klog.Errorf("Smooth migration for tm[%s/%s], fail to patch PV %s, err: %v", monitor.Namespace, monitor.Name, deploymentPv.Name, err)
+								return false, err
+							}
+							// smooth migration successfully and clean status
+							monitor.Status.DeploymentStorageStatus = nil
+						}
+						return true, nil
+					}
+					klog.Errorf("Smooth migration for tm[%s/%s], the PVC of the deployment get error:%v", monitor.Namespace, monitor.Name, err)
+					return false, err
+				}
+
+				// firstly patch status
+				if monitor.Status.DeploymentStorageStatus == nil {
+					monitor.Status.DeploymentStorageStatus = &v1alpha1.DeploymentStorageStatus{
+						PvName: deploymentPvc.Spec.VolumeName,
+					}
+					err = m.deps.TypedControl.UpdateStatus(monitor)
```

Comment:
This will erase the other fields in the status, you can patch the status similar with what we're doing in https://github.com/pingcap/tidb-operator/blob/master/pkg/monitor/monitor/monitor_manager.go#L397